### PR TITLE
Add support for the NUCLEO-WL55JCx

### LINKED ISSUE #543: Board support request: NUCLEO-WL55JCx
The board comes in two versions NUCLEO-WL55JC1, and NUCLEO-WL55JC2, where the differences is what frequency they support for the radio: https://www.st.com/en/evaluation-tools/nucleo-wl55jc.html

The board is already supported by Mbed (https://os.mbed.com/platforms/ST-Nucleo-WL55JC/), and Zephyr (https://docs.zephyrproject.org/latest/boards/arm/nucleo_wl55jc/doc/nucleo_wl55jc.html).

I have access to both versions of the board. I have managed to get them working in PlatformIO with the Mbed framework, by adding a custom "board.json" file to my boards directory.

Status:
* I have only tested STlink
* I have only tested Mbed
* I am not sure how to obtain the .svd file
* The two versions of the board requires different configuration for the LoRaWan frequency. I am not sure if this should be accounted for (e.g. by having two different board configurations), or if it should be the responsibility of the user to figure this out.

```
{
  "build": {
    "core": "stm32",
    "cpu": "cortex-m4",
    "extra_flags": "-DSTM32WL55JC1",
    "f_cpu": "48000000L",
    "mcu": "stm32wl55jc",
    "product_line": "STM32WL55xx",
    "variant": "STM32WL55JCx/WL55JC(1-2)"
  },
  "debug": {
    "default_tools": [
      "stlink"
    ],
    "jlink_device": "STM32WL55JC",
    "onboard_tools": [
      "stlink"
    ],
    "openocd_target": "stm32wlx",
    "svd_path": "STM32F411xx.svd"
  },
  "frameworks": [
    "arduino",
    "cmsis",
    "mbed",
    "stm32cube",
    "libopencm3",
    "zephyr"
  ],
  "name": "ST Nucleo WL55JC",
  "upload": {
    "maximum_ram_size": 65536,
    "maximum_size": 262144,
    "protocol": "stlink",
    "protocols": [
      "jlink",
      "cmsis-dap",
      "stlink",
      "blackmagic",
      "mbed"
    ]
  },
  "url": "https://www.st.com/en/evaluation-tools/nucleo-wl55jc.html",
  "vendor": "ST"
}
```

I'd be happy to create a pull request as soon as I have figured out how to resolve the remaining issues.

Is there anything else that should be updated while adding a new board? Do I create a separate pull request for updating the documentation?

### PR DESCRIPTION
**Pull request by gbThreepwood**
Added support for the NUCLEO-WL55JCx boards. Resolved #543

### CODE REVIEW

File: boards/nucleo_wl55jc.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "svd_path": "STM32WL5x_CM4.svd"
+  },
+  "frameworks": [
+    "arduino",
+    "cmsis",
+    "mbed",
+    "stm32cube",
+    "libopencm3",
+    "zephyr"
+  ],
```
Comment (valeros) [inline boards/nucleo_wl55jc.json:29]: Please review the framework list.
- It seems there is no support for this board in the latest stable Arduino v2.0, probably will be added in v2.1.
- CMSIS and STM32Cube are not supported at the moment as well.
- libOpenCM3 doesn't have support for the `WL` series.

So I guess the only option left is Zephyr.

File: boards/nucleo_wl55jc.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  "platforms": ["ststm32"],
```
Comment (valeros) [inline boards/nucleo_wl55jc.json:30]: Seems redundant to me. It's set automatically if the board in located in the platform.

File: boards/nucleo_wl55jc.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  "name": "ST Nucleo WL55JC",
+  "upload": {
+    "maximum_ram_size": 65536,
+    "maximum_size": 262144,
+    "protocol": "stlink",
+    "protocols": [
+      "jlink",
+      "cmsis-dap",
+      "stlink",
+      "blackmagic",
```
Comment (valeros) [inline boards/nucleo_wl55jc.json:40]: It seems the `blackmagic` debug probe doesn't support the WL series.

**Comment by valeros** [comment]
Hi @gbThreepwood ! Many thanks for the PR. Please review my comments.

**Review by valeros — CHANGES_REQUESTED** [review]

File: boards/nucleo_wl55jc.json (hunk 2 above)
Comment (gbThreepwood) [reply boards/nucleo_wl55jc.json]: I see. I saw this line in one of the json files I used for reference, guess I just assumed it had to be there.

File: boards/nucleo_wl55jc.json (hunk 3 above)
Comment (gbThreepwood) [reply boards/nucleo_wl55jc.json]: Yes. Just to make sure I also had a look now, and unfortunately it looks like you are right in that they do not support this board.

File: boards/nucleo_wl55jc.json (hunk 1 above)
Comment (gbThreepwood) [reply boards/nucleo_wl55jc.json]: Removed everything except Mbed, and Zephyr.

Comment (valeros) [reply boards/nucleo_wl55jc.json]: It seems mbed is not available as well. At least in v6.9 which we support at the moment. I think it's reasonable to remove it for now, until the latest mbed is added to the platform.

Comment (gbThreepwood) [reply boards/nucleo_wl55jc.json]: Removed. I am using the board successfully with Mbed 6.9, so I did not even think to check if it was officially supported.

**Comment by valeros** [comment]
Thanks for quick fixes, merged.

**Pull request merged**